clients/src/main/java/org/apache/kafka/common/requests/WriteTxnMarkersRequest.java

```diff
@@ -183,4 +211,17 @@ public static WriteTxnMarkersRequest parse(ByteBuffer buffer, short version) {
         return new WriteTxnMarkersRequest(ApiKeys.WRITE_TXN_MARKERS.parseRequest(version, buffer), version);
     }
 
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        final WriteTxnMarkersRequest that = (WriteTxnMarkersRequest) o;
+        return coordinatorEpoch == that.coordinatorEpoch &&
```

Comment:
So this should change back to what you previously had? We originally had your code, but during the merge with other changes it was probably removed. That is why i did the groupBy on `coordinatorEpoch`.